apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -429,6 +435,98 @@ func (s *StringTransform) Resolve(input interface{}) (interface{}, error) {
 	return fmt.Sprintf(s.Format, input), nil
 }
 
+// The list of supported ConvertTransform input and output types.
+const (
+	ConvertTransformTypeString  = "string"
+	ConvertTransformTypeBool    = "bool"
+	ConvertTransformTypeInt     = "int"
+	ConvertTransformTypeFloat64 = "float64"
+)
+
+var (
+	conversions = map[string]func(interface{}) (interface{}, error){
```

Comment:
I went back and forth between having this map or nested `switch` statements for pairs. I feel like this is easier to write tests for and doesn't increase the cyclomatic complexity of the `Resolve` function like `switch` does (though we'd skip since it's not really more complex). You can see the `switch` structure in the earlier commit.

Reply:
`reflect.Kind` is just an `iota` and to get the actual string, we use `.String()` method. However, it doesn't work backwards, i.e. there is no function to take `int` and convert it to `reflect.Kind`. We can implement it but `reflect` doesn't expose the array so we'd have to duplicate those bindings, too. So, I just opted for using string as type for those fields:
```golang
type conversionPair struct {
  From string
  To string
}
```

Reply:
>  i.e. there is no function to take int and convert it to reflect.Kind

Ah, this confused me for a bit but I think you're saying there's no way to take the string `"int"` and convert it to a `reflect.Kind`, right? I was initially thinking you were saying it wasn't possible to get the `reflect.Kind` of an `int`.